mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorAllocationHelper.java

```diff
@@ -0,0 +1,152 @@
+            final String constraintValue = constraints.getAssignmentAttributes().getOrDefault(e.getKey(), e.getValue());
+            if (!skuAttrValue.equalsIgnoreCase(constraintValue)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private ContainerSkuID findBestFit(ContainerSkuID sku0, ContainerSkuID sku1) {
+        final SkuSizeSpec size0 = skuSizes.get(getSkuSizeIdFrom(skus.get(sku0)));
+        final SkuSizeSpec size1 = skuSizes.get(getSkuSizeIdFrom(skus.get(sku1)));
+        if (size0.getCpuCoreCount() < size1.getCpuCoreCount() || (
+            size0.getCpuCoreCount() == size1.getCpuCoreCount() && size0.getMemorySizeInMB() < size1.getMemorySizeInMB()
+        )) {
+            return sku0;
+        }
+        return sku1;
+    }
+
+    private void refreshMetadata(ContainerSkuID skuID) {
+        try {
+            Objects.requireNonNull(this.storageProvider.getResourceClusterSpecWritable(clusterID))
+                .getClusterSpec()
+                .getSkuSpecs()
+                .stream()
+                .filter(s -> s.getSkuId().equals(skuID))
+                .forEach(skuTypeSpec -> {
+                    this.skus.put(skuID, skuTypeSpec);
+
+                    // If skuSizeID hasn't been set yet --> create one on the fly
+                    // TODO: remove this when all sku specs are updated with sizeID
+                    if (skuTypeSpec.getSizeId() == null) {
+                        SkuSizeSpec skuSizeSpec = SkuSizeSpec.builder()
+                            .skuSizeID(getSkuSizeIdFrom(skuTypeSpec))
+                            .skuSizeName(getSkuSizeIdFrom(skuTypeSpec).getResourceID())
+                            .cpuCoreCount(skuTypeSpec.getCpuCoreCount())
+                            .memorySizeInMB(skuTypeSpec.getMemorySizeInMB())
+                            .diskSizeInMB(skuTypeSpec.getDiskSizeInMB())
+                            .networkMbps(skuTypeSpec.getNetworkMbps())
+                            .build();
+                        this.skuSizes.put(skuSizeSpec.getSkuSizeID(), skuSizeSpec);
+                        return;
+                    }
+                    if (!skuSizes.containsKey(skuTypeSpec.getSizeId())) {
+                        try {
+                            this.storageProvider.getResourceClusterSkuSizes()
+                                .forEach(skuSizeSpec -> this.skuSizes.put(skuSizeSpec.getSkuSizeID(), skuSizeSpec));
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private SkuSizeID getSkuSizeIdFrom(SkuTypeSpec skuTypeSpec) {
+        return skuTypeSpec.getSizeId() == null ? SkuSizeID.of("skuTypeSpec-" + skuTypeSpec.getSkuId().getResourceID()): skuTypeSpec.getSizeId();
+    }
```

Comment:
This logic could be moved to SkuTypeSpec to provide a standard way of handling it, instead of being deeply baked into assumptions.